### PR TITLE
Fix  GSA bugs

### LINKED ISSUE #710: Tweak Ground Spike Antenna
- ~Anyone can disconnect GSA radio (of other person)~ -> intentional
- [X] Add notification if no space to pick-up (can get dropped on floor without noticing)
- [X] Add notification on disconnect (especially when someone else disconnects you)
- [ ] Fix interaction point on Mast being offset from the box
- [ ] Improve placement when mounting/demounting Mast (gets displaced by a few meters)

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Partially fixes #710 
  - Add notification if no space to pick-up (can get dropped on floor without noticing)
  - Add notification on disconnect (especially when someone else disconnects you)
- Fixes locality of disconnect. Everything was supposed to run on the server but it was executed on the client.
- Make pick-up function more robust: fixes a bug where the spike could not be picked up if the player had a mast and not so much space.
- Do not show actions to connect, if no radios can  be connected to the GSA.

### CODE REVIEW

File: addons/sys_gsa/XEH_postInit.sqf [hunk 1]
```diff
@@ -10,6 +10,12 @@
         _success
     };
 
+    private _connectedUnit = [_radioId] call EFUNC(sys_radio,getRadioObject);
+    if (isNull _connectedUnit) exitWith {
+        ERROR("Null connected object returned");
```
Comment (jonpas) [inline addons/sys_gsa/XEH_postInit.sqf:15]: This does a pop-up.

File: addons/sys_gsa/fnc_hasCompatibleRadios.sqf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Public: No
+ */
+
+ params [["_unit", objNull], ["_gsa", objNull]];
+
+if (isNull _unit || {isNull _gsa}) exitWith {
+    false
+};
+
+private _radioList = [] call EFUNC(sys_data,getPlayerRadioList);
```
Comment (jonpas) [inline addons/sys_gsa/fnc_hasCompatibleRadios.sqf:25]:
```suggestion
private _radioList = [] call EFUNC(sys_data,getPlayerRadioList);
_radioList findIf {[_gsa, _x] call FUNC(isRadioCompatible)} != -1
```
Everything below can be removed. :)

File: addons/sys_gsa/fnc_pickUp.sqf [hunk 3]
```diff
@@ -33,21 +33,28 @@ switch (_classname) do {
...
         if (_unit canAdd _item) then {
-            _item = _item;
-            _canDelete = true;
+            // If the unit has a mast, delete the mast item and try to add the whole antenna. Usually this should
+            // not happen if the masses are consistent, but better be sure.
+            if ([_unit, "ACRE_VHF30108MAST"] call EFUNC(sys_core,hasItem)) then {
+                _unit removeItem "ACRE_VHF30108MAST";  // Delete the mast
+                if (_unit canAdd "ACRE_VHF30108") then {
+                    _item = "ACRE_VHF30108";
+
```
Comment (jonpas) [inline addons/sys_gsa/fnc_pickUp.sqf]:
```suggestion
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_gsa/XEH_postInit.sqf (hunk 1 above)
Comment (TheMagnetar) [reply addons/sys_gsa/XEH_postInit.sqf:15]: It was not doing a pop-up if radio is empty...

Comment (jonpas) [reply addons/sys_gsa/XEH_postInit.sqf:15]: Ah, that may be `ERROR_MSG` then that I am thinking about.

Comment (TheMagnetar) [reply addons/sys_gsa/XEH_postInit.sqf:15]: yes. this only writes ERROR (XXX): in the rpt

**Comment by TheMagnetar** [comment]
No

**Comment by jonpas** [comment]
Why is it necessary, what command are you waiting for to finish?

**Review by jonpas — APPROVED** [review]

**Pull request merged**